### DISCUSSION TITLE
select subset of rows via non boolean expression

### DISCUSSION THREAD

**Discussion by san-vak** [vaexio/vaex#2277] [General]
Hi
thanks for the great library

i was wondering if there is a way to select a subset of rows via some non boolean expression
like in this example
`new_df = df.select(df["sub_id"] in id_list)`

i tried to use this, and it only returns the boolean result (True/False)
`new_df = df[~(for i in df["subid"] if i in id_list)]`

kind regards

**Comment by san-vak** [comment]
so i could get the results that i wanted via:

```
subid_df = df["subid"].isin(id_list)
indecies = []
for index in range(len(subid_df)):
    if subid_df[index] == True:
        indecies.append(index)
        
new_df = df.take(indecies)
```
i was wondering if there is a simpler way?

best wishes

**Reply by JovanVeljanoski** [reply]
Hi, 

You were very very close :)

If i understand your need correctly, then you can simply do 

```python
df_filtered = df[df["subid"].isin(id_list)]
```

or in a step by step way

```python
bool_expr = df["subid"].isin(id_list)
df_filtered = df[bool_expr]
```


Does this make sense?

**Reply by san-vak** [reply]
yes!!
thank you so much
